src/app/features/task-repeat-cfg/store/task-repeat-cfg.reducer.spec.ts

```diff
@@ -195,71 +224,275 @@ describe('selectTaskRepeatCfgsDueOnDay', () => {
-    //         repeatEvery: 2,
-    //       }),
-    //     ],
-    //     {
-    //       dayDate: new Date('2022-03-10').getTime(),
-    //     },
-    //   );
-    //   const resultIds = result.map((item) => item.id);
-    //   expect(resultIds).toEqual(['R1']);
-    // });
-    // it('should NOT return cfg if repeatCycle does NOT match', () => {
-    //   const result = selectTaskRepeatCfgsDueOnDay.projector(
-    //     [
-    //       dummyRepeatable('R1', {
-    //         repeatCycle: 'WEEKLY',
-    //         startDate: '2022-01-10',
-    //         repeatEvery: 3,
-    //       }),
-    //     ],
-    //     {
-    //       dayDate: new Date('2022-03-10').getTime(),
-    //     },
-    //   );
-    //   const resultIds = result.map((item) => item.id);
-    //   expect(resultIds).toEqual([]);
-    // });
+      const dayDate = new Date('2022-01-11'); // Tuesday
+      // @ts-ignore
+      const printDayDate = dayDate.toLocaleTimeString('en-us', datePrintingOptions);
+
+      const result = selectTaskRepeatCfgsDueOnDay.projector(
+        [
+          dummyRepeatable('R1', {
+            repeatCycle: 'WEEKLY',
+            startDate: '2022-01-12',
+            tuesday: true,
+          }),
+        ],
+        {
+          dayDate: dayDate.getTime(),
+        },
+      );
+      const resultIds = result.map((item) => item.id);
+
+      console.log(
+        '\nIDs: ',
+        resultIds,
+        '\nStart: ',
+        printStart,
+        '\nToday:',
+        printDayDate,
+      );
+      expect(resultIds).toEqual([]);
+    });
+    it('should return cfg if repeatCycle matches', () => {
+      const result = selectTaskRepeatCfgsDueOnDay.projector(
+        [
+          dummyRepeatable('R1', {
+            monday: true,
+            repeatCycle: 'WEEKLY',
+            startDate: '2022-01-10',
+            repeatEvery: 2,
+          }),
+        ],
+        {
+          //dayDate: new Date('2022-03-10').getTime(),
+          dayDate: new Date('2022-01-24').getTime(),
+        },
+      );
+      const resultIds = result.map((item) => item.id);
+      console.log(resultIds);
+      expect(resultIds).toEqual(['R1']);
+    });
+    it('should NOT return cfg if repeatCycle does NOT match', () => {
+      const result = selectTaskRepeatCfgsDueOnDay.projector(
+        [
+          dummyRepeatable('R1', {
+            monday: true,
+            repeatCycle: 'WEEKLY',
+            startDate: '2022-01-10',
+            repeatEvery: 3,
+          }),
+        ],
+        {
+          dayDate: new Date('2022-01-17').getTime(),
+        },
+      );
+      const resultIds = result.map((item) => item.id);
+      expect(resultIds).toEqual([]);
+    });
+
+    // Multiple Weeks Test
+    // Please keep in mind, this test triggers every day of the week. Hence, it does not test the behaviour, if a day is skipped.
+    // For convenience of the test, lastTaskCreation is always the day before the day we test.
+    const expectedMultipleTest = {
```

Comment:
Something like 
```
[ 
  [condition, result],
  [condition, result], 
  [condition, result]
].forEach(([condtion,result]) => it('....'))
```  might improve readability.

Reply:
Could you explain the enhancement of this change a bit further? I don't understand the benefit. – What is the `condition` in your change?
_(Using more complex constructions doesn't improve readability, IMHO.)_